wetmap/src/components/newModals/tripCreatorListModal/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { itineraries } from '../../../supabaseCalls/itinerarySupabaseCalls';
+import { updateDiveShop } from '../../../supabaseCalls/shopsSupabaseCalls';
+import { SelectedShopContext } from '../../contexts/selectedShopContext';
+import { UserProfileContext } from '../../contexts/userProfileContext';
+import { ModalContext } from '../../reusables/modal/context';
+import { ItineraryItem } from './types';
+import TripCreatorListView from './view';
+import { ModalHandleProps } from '../../reusables/modal/types';
+import DiveShopModal from '../shopModal/index';
+
+type TripCreatorListProps = Partial<ModalHandleProps>;
+
+export default function TripCreatorListModal(props: TripCreatorListProps) {
+  const { selectedShop } = useContext(SelectedShopContext);
+  const { profile } = useContext(UserProfileContext);
+  const modalContext = useContext(ModalContext);
+  const { modalShow } = useContext(ModalContext);
+
+  const [itineraryList, setItineraryList] = useState<ItineraryItem[]>([]);
+  const [selectedID, setSelectedID] = useState<number>(0);
+
+  useEffect(() => {
+    if (selectedShop) {
+      getItineraries(selectedShop.id);
+    }
+  }, [selectedShop]);
+
+  const getItineraries = async (IdNum: number) => {
```

Comment:
hmmm, we have used this exact code to get the itineraries elsewhere in the project (Dive Shop Modal), maybe we should move this into a reusable and simply import it on pages that need it rather than having this code block repeated on them.

Reply:
Thinking of saving this part for the PR where we revamp the itinerary file and object

Reply:
ok sounds good